Scripts/SL-Helpers.lua

```diff
@@ -988,4 +988,48 @@ GetPlayerOptionsString = function(player, modsLevel)
 	end
 
 	return optionslist
+end
+-- -----------------------------------------------------------------------
+-- helper function for returning the player AF
+-- works as expected in ScreenGameplay
+--     arguments:  pn is short string PlayerNumber like "P1" or "P2"
+--     returns:    the "PlayerP1" or "PlayerP2" ActorFrame in ScreenGameplay
+--                 or, the unnamed equivalent in ScrenEdit
+GetPlayerAF = function(pn)
+	local topscreen = SCREENMAN:GetTopScreen()
+	if not topscreen then
+		lua.ReportScriptError("GetPlayerAF() failed to find the player ActorFrame because there is no Screen yet.")
+		return nil
+	end
+
+	local playerAF = nil
+
+	-- Get the player ActorFrame on ScreenGameplay
+	-- It's a direct child of the screen and named "PlayerP1" for P1
+	-- and "PlayerP2" for P2.
+	-- This naming convention is hardcoded in the SM5 engine.
+	--
+	-- ScreenEdit does not name its player ActorFrame, but we can still find it.
+
+	-- find the player ActorFrame in edit mode
+	if (THEME:GetMetric(topscreen:GetName(), "Class") == "ScreenEdit") then
+		-- loop through all nameless children of topscreen
+		-- and find the one that contains the NoteField
+		-- which is thankfully still named "NoteField"
+		for _,nameless_child in ipairs(topscreen:GetChild("")) do
+			if nameless_child:GetChild("NoteField") then
+				playerAF = nameless_child
+				break
```

Comment:
In Versus practice mode you need to gather all the playerAFs you find and pick the correct one to return based on pn, vs returning just the first one here.

Reply:
That does explain some of the behavior I was seeing, was about to reply with a comment of a few issues...

That being said, is there a way to get the pn from the Notefield or Player? Looking at the reference I'm not seeing much there, so maybe we consider naming the the player actor frames in the engine?

Reply:
For reference, I just spun this up in a debugger with naming the actor frame and removing the edit mode specific code from here, and it works.

Reply:
Changed it to use a common path, though it depends on this PR being included in engine: https://github.com/itgmania/itgmania/pull/651